src/include/planner/subplans_table.h

```diff
@@ -13,30 +13,43 @@ namespace kuzu {
 namespace planner {
 
 const uint64_t MAX_LEVEL_TO_PLAN_EXACTLY = 7;
-const uint64_t MAX_NUM_SUBGRAPHS_PER_LEVEL = 100;
+const uint64_t MAX_NUM_SUBGRAPHS_PER_LEVEL = 50;
 const uint64_t MAX_NUM_PLANS_PER_SUBGRAPH = 50;
 
 class SubPlansTable {
-    typedef std::unordered_map<SubqueryGraph, std::vector<std::unique_ptr<LogicalPlan>>,
-        SubqueryGraphHasher>
-        SubqueryGraphPlansMap;
+    struct PlanSet;
+    // Each db level is a map from sub query graph to a set of plans
+    using dp_level_t = sub_query_graph_map_t<std::unique_ptr<PlanSet>>;
 
 public:
     void resize(uint32_t newSize);
 
     bool containSubgraphPlans(const SubqueryGraph& subqueryGraph) const;
 
-    std::vector<std::unique_ptr<LogicalPlan>>& getSubgraphPlans(const SubqueryGraph& subqueryGraph);
+    logical_plan_vector_t& getSubgraphPlans(const SubqueryGraph& subqueryGraph);
 
     std::vector<SubqueryGraph> getSubqueryGraphs(uint32_t level);
 
     void addPlan(const SubqueryGraph& subqueryGraph, std::unique_ptr<LogicalPlan> plan);
-    void finalizeLevel(uint32_t level);
 
     void clear();
 
 private:
-    std::vector<std::unique_ptr<SubqueryGraphPlansMap>> subPlans;
+    struct PlanSet {
+        logical_plan_vector_t plans;
+        schema_map_t<common::vector_idx_t> schemaToPlanIdx;
```

Comment:
This is the part i'm a bit confused. 
```
        logical_plan_vector_t plans;
        schema_map_t<common::vector_idx_t> schemaToPlanIdx;
```
Why not just keep a map from schema to logical plan? so you don't need to keep a vector separately.

Reply:
because I need to return plans as a vector